CalendarApp/Logic Handlers/SyncConflictHandler.swift

```diff
@@ -7,13 +7,204 @@
 
 import Foundation
 
+@objc
+public protocol SyncChangesDelegate {
+    func deletedEventOnRemote(eventID: UUID)
+    func updatedEventOnRemote(event: Event)
+    func createdEventOnRemote(eventID: UUID)
+    
+    func syncNewEventToParse(event: Event, remoteChange: RemoteChange)
+    func syncDeleteToParse(event: String, remoteChange: RemoteChange)
+    func syncUpdateToParse(event: Event)
+    func syncRemoteChangeToParse(remoteChange: RemoteChange)
+}
+
 @objcMembers
 class SyncConflictHandler : NSObject {
-    private var keptChanges = [LocalChange]()
+    weak var delegate: SyncChangesDelegate?
+    private var revisionHistories: [[Revision]]
+    private let localChangeHandler = LocalChangeHandler()
+    private let parseChangeHandler = ParseChangeHandler()
+    private let coreDataEventHandler = CoreDataEventHandler()
+    
+    init(histories: [[RemoteChange]]) {
+        revisionHistories = histories
```

Comment:
just a suggestion, you don't have to change it, but you can also call `revisionHistories` `histories` and do `self. histories = histories`. The name of an instance property can be the same as the init parameter.